geostructures/parsers.py

```diff
@@ -24,6 +25,242 @@
 }
 
 
+def _get_datetime_pandas(start_time, end_time):
+    """
+    Converts pandas Timestamps to Python datetime objects and returns a TimeInterval.
+
+    Args:
+        start_time (pd.Timestamp or None): The start time.
+        end_time (pd.Timestamp or None): The end time.
+
+    Returns:
+        TimeInterval: The time interval representing the start and end time.
+    """
+    import pandas as pd
+    from geostructures.time import TimeInterval
+
+    if pd.notnull(start_time) or pd.notnull(end_time):
+        if isinstance(start_time, pd.Timestamp):
+            start_time = start_time.to_pydatetime()
+
+        if isinstance(end_time, pd.Timestamp):
+            end_time = end_time.to_pydatetime()
+
+        return TimeInterval(start_time, end_time)
+
+
+def parse_arcgis_featureclass(
+    sedf,
+    time_start_property: Optional[str] = None,
+    time_end_property: Optional[str] = None,
+    _shapes: Optional[List[GeoShape]] = None,
+    _props: Optional[Dict[str, str]] = None,
+):
+    """
+    Parses a Spatially Enabled DataFrame (SEDF) from ArcGIS and converts it into geostructures.
+
+    Args:
+        sedf (DataFrame):
+            A Spatially Enabled DataFrame containing feature class data.
+
+        time_start_property (str, optional):
+            The name of the field containing the start time data.
+
+        time_end_property (str, optional):
+            The name of the field containing the end time data.
+
+        _shapes (List[GeoShape], optional):
+            Internal use only. Mutated with geostructures as they're extracted from the feature class.
+
+        _props (Dict[str, str], optional):
+            Internal use only. Information about higher-level containers
+            to store as properties on the shape for traceability.
+
+    Returns:
+        List[GeoShape]: A list of GeoShape objects parsed from the feature class.
+    """
+    if _shapes is None:
+        _shapes = []
+    if _props is None:
+        _props = {}
+
+    property_columns = [col for col in sedf.columns if col != 'SHAPE']
+
+    for row in sedf.itertuples():
+        geometry = row.SHAPE
+        geometry_type_str = type(geometry).__name__.upper()
+        if geometry_type_str not in _PARSER_MAP:
+            raise ValueError(f"Unsupported geometry type: {geometry_type_str}")
+
+        parser = _PARSER_MAP[geometry_type_str]
+        properties = {col: getattr(row, col) for col in property_columns}
+
+        if geometry_type_str == 'POINT':
+            lon, lat = geometry.x, geometry.y
+            shape = parser(Coordinate(lon, lat), properties=properties)
```

Comment:
It looks like you're using `__init__` - let's talk about this, I think we can simplify things a little